controllers/machine_controller.go

```diff
@@ -339,32 +350,26 @@ func (r *MachineReconciler) drainNode(cluster *clusterv1.Cluster, nodeName strin
 
 func (r *MachineReconciler) deleteNode(ctx context.Context, cluster *clusterv1.Cluster, name string) error {
 	logger := r.Log.WithValues("machine", name)
-
+	var c client.Client
+	var err error
 	if cluster == nil {
 		// Try to retrieve the Node from the local cluster, if no Cluster reference is found.
-		var node corev1.Node
-		if err := r.Client.Get(ctx, client.ObjectKey{Name: name}, &node); err != nil {
-			return err
+		c = r.Client
```

Comment:
We might need to open an issue about this though, because after #1539 the Cluster is guaranteed to be there, so we'll never get to use the local client

Reply:
Should we open an issue to audit the code base to find everywhere we check for a nil Cluster?

Reply:
Yeah, we probably should. I'd also like to explore an option where we can detect (and document how) the controllers are running in the same cluster that's being managed

Reply:
That would be ideal. Not sure there's anything generic, though?

Reply:
Opened https://github.com/kubernetes-sigs/cluster-api/issues/1666 to discuss

Reply:
And https://github.com/kubernetes-sigs/cluster-api/issues/1667 for the cluster checks